internal/configs/virtualserver_test.go

```diff
@@ -1361,6 +1361,357 @@ func TestGenerateVirtualServerConfigForVirtualServerWithMatches(t *testing.T) {
 	}
 }
 
+func TestGenerateVirtualServerConfigForVirtualServerRoutesWithDos(t *testing.T) {
+	dosResources := map[string]*appProtectDosResource{
+		"/coffee": {
+			AppProtectDosEnable:          "on",
+			AppProtectDosLogEnable:       false,
+			AppProtectDosMonitorUri:      "test.example.com",
+			AppProtectDosMonitorProtocol: "http",
+			AppProtectDosMonitorTimeout:  0,
+			AppProtectDosName:            "my-dos",
+			AppProtectDosAccessLogDst:    "svc.dns.com:123",
+			AppProtectDosPolicyFile:      "",
+			AppProtectDosLogConfFile:     "",
+		},
+	}
+
+	virtualServerEx := VirtualServerEx{
+		VirtualServer: &conf_v1.VirtualServer{
+			ObjectMeta: meta_v1.ObjectMeta{
+				Name:      "cafe",
+				Namespace: "default",
+			},
+			Spec: conf_v1.VirtualServerSpec{
+				Host: "cafe.example.com",
+				Upstreams: []conf_v1.Upstream{
+					{
+						Name:    "tea-v1",
+						Service: "tea-svc-v1",
+						Port:    80,
+					},
+					{
+						Name:    "tea-v2",
+						Service: "tea-svc-v2",
+						Port:    80,
+					},
+				},
+				Routes: []conf_v1.Route{
+					{
+						Path: "/tea",
+						Matches: []conf_v1.Match{
```

Comment:
because we don't test DOS generation for VS in this test, we can remove the upstreams, the tea endpoints and the `/tea` route from this test - this will greatly reduce the amount of the code.